desc/magnetic_fields/_current_potential.py

```diff
@@ -842,3 +1042,790 @@ def nfp_loop(j, f):
+    surface current (B_Phi_SV for field from the single valued part Φₛᵥ, and
+    B_GI for that from the secular terms I and G), plasma current, and external
+    fields::
+
+        Bn = ∫ ∫ (B . n)^2 dA
+        B = B_plasma + B_external + B_Phi_SV + B_GI
+
+    G is fixed by the equilibrium magnetic field strength, and I is determined
+    by the desired coil topology (given by ``current_helicity``), with zero
+    helicity corresponding to modular coils, and non-zero helicity corresponding
+    to helical coils. The algorithm then finds the single-valued part of Φ
+    by minimizing the quadratic flux on the plasma surface along with a
+    regularization term on the surface current magnitude::
+
+        min_Φₛᵥ  ∫ ∫ (B . n)^2 dA + λ ∫ ∫ | K |^2 dA
+
+    where λ is the regularization parameter, smaller `lambda_regularization`
+    corresponds to less regularization (consequently, lower Bn error but more
+    complex and large surface currents) and larger `lambda_regularization`
+    corresponds to more regularization (consequently, higher Bn error but simpler
+    and smaller surface currents).
+
+    [1] Landreman, An improved current potential method for fast computation
+        of stellarator coil shapes, Nuclear Fusion (2017)
+
+    Parameters
+    ----------
+    current_potential_field : FourierCurrentPotentialField
+        ``FourierCurrentPotentialField`` to run REGCOIL with.
+    eq : Equilibrium
+        Equilibrium to minimize the quadratic flux (plus regularization) on.
+    lambda_regularization : float or ndarray, optional
+        regularization parameter, > 0, regularizes minimization of Bn
+        on plasma surface with minimization of current density mag K on winding
+        surface i.e. larger lambda_regularization, simpler coilset and smaller
+        currents, but worse Bn. If a float, only runs REGCOIL for that single value
+        and returns a single FourierCurrentPotentialField and the associated data.
+        If an array is passed, will run REGCOIL for each lambda_regularization in
+        that array and return a list of FourierCurrentPotentialFields, and the
+        associated data.
+    source_grid : Grid, optional
+        Source grid upon which to evaluate the surface current when calculating
+        the normal field on the plasma surface. Also used to evaluate the
+        virtual casing current, if the plasma has finite plasma currents.
+        Defaults to
+        LinearGrid(M=max(3 * current_potential_field.M_Phi, 30),
+        N=max(3 * current_potential_field.N_Phi, 30), NFP=eq.NFP)
+    eval_grid : _type_, optional
+        Grid upon which to evaluate the normal field on the plasma surface, and
+        at which the normal field is minimized.
+        Defaults to
+        `LinearGrid(M= 30, N= 30, NFP=eq.NFP)`
+    current_helicity : tuple of size 2, optional
+        Tuple of (q,p) used to determine coil topology, where q is the
```

Comment:
https://www.nature.com/articles/s41598-019-53548-w

in my paper I follow the notation of this reference, I could change it I suppose, will just need to reverse it in my paper as well and note I am making that choice. I made it (q,p) here to make it more like (m,n), since to me q and p arent used in the context of symmetries usually so I felt there wasnt a connotation of poloidal for p or anything like that

Reply:
I will make this change then, ok, and try to document well here exactly what they mean.